### DISCUSSION TITLE
Curious about `vercel.json` clearnUrls & rewrites options

### DISCUSSION THREAD

**Discussion by 2ssue** [vercel/vercel#6694] [Help] [answered]
I deployed a static website using Vercel, and I wanted to use both Vercel configuration `cleanUrls` & `rewrites`. 
So I made it like below, but I got '404' when access on `/asdf/1234` and so on.
```json
{
    "cleanUrls": true,
    "rewrites": [
        {"source": "/asdf/:asdf*", "destination": "/index.html"}
    ]
}
```

But when I removed `cleanUrls` option, I got the expected page, `index.html`. So I want to know why is this work. Are these options conflicting? Why does clearing this option work?

My repository is structured like this:
```bash
📁root
├ 📁 assets (fonts/imgs)
├ 📁 styles (css)
├ 📄index.html
├ 📄error.html
⎣ 📄index.js
```

**Comment by delbaoliveira** [comment, marked as answer]
Hey @2ssue,

Could you try removing the `.html` from the destination:

 `{"source": "/asdf/:asdf*", "destination": "/index"}`

**Reply by 2ssue** [reply]
OMG, thanks!! I didn't know cleanUrls option has a higher priority...

Where could I find out about this order of execution? I thought rewrites work first because of cleanUrls option explains redirect to clean upped URL.

**Reply by delbaoliveira** [reply]
Awesome, glad it worked! We don't explicitly state it in our docs but it's something we should consider 👍🏼 

cleanURLs execute at build time so the file extensions are _'removed'_ by the time rewrites are processed during production [at the Edge](https://vercel.com/docs/cli#project-configuration/rewrites). This explains why we get the 404 error if we use file extensions inside **rewrites** while also using **cleanURLs**.

**Reply by 0xNegro** [reply]
How would someone go about doing this on a vercel maintained deployment? I'm guessing this is all being done on the command line. But my index.html is also not being read.
